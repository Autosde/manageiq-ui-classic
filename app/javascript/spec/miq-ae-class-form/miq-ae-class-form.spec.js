import React from 'react';
import fetchMock from 'fetch-mock';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MiqAeClass from '../../components/miq-ae-class';

describe('MiqAeClass Form Component', () => {
  const classMockData = [
    {
      href: `/miq_ae_class/edit_class/2/`,
      id: 2,
      description: 'Configured System Provision',
    },
  ];

  const MiqAeClassEditData = {
    id: 40,
    name: 'test',
    display_name: 'test display name',
    description: 'test description',
  };

  const fqName = 'Sample FQ Name';

  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
  });

  it('should render add class form correctly', async() => {
    const wrapper = shallow(<MiqAeClass
      classRecord={undefined}
      fqname={fqName}
    />);

    fetchMock.get(`/miq_ae_class/new?&expand=resources/`, classMockData);

    await new Promise((resolve) => {
      setImmediate(() => {
        wrapper.update();
        expect(toJson(wrapper)).toMatchSnapshot();
        resolve();
      });
    });
  });

  it('should render edit class form correctly', async() => {
    const wrapper = shallow(<MiqAeClass
      classRecord={MiqAeClassEditData}
      fqname={fqName}
    />);

    fetchMock.get(`/miq_ae_class/edit_class_react/${MiqAeClassEditData.id}?&expand=resources/`, classMockData);
    await new Promise((resolve) => {
      setImmediate(() => {
        wrapper.update();
        expect(toJson(wrapper)).toMatchSnapshot();
        resolve();
      });
    });
  });
});
