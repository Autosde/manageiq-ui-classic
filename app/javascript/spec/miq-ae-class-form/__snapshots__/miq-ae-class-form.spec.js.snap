// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiqAeClass Form Component should render add class form correctly 1`] = `""`;

exports[`MiqAeClass Form Component should render edit class form correctly 1`] = `""`;
